patches/0005-Prefer-Quilt-over-Fabric-Loader.patch

```diff
@@ -5,66 +5,46 @@ Subject: [PATCH] Prefer Quilt over Fabric Loader
  import com.google.gson.GsonBuilder;
 +
 +import net.fabricmc.loom.util.FabricCapabilities;
 +
  import org.gradle.api.Project;
++import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
  import org.gradle.api.plugins.PluginAware;
  
-@@ -83,6 +86,9 @@ public class LoomGradlePlugin implements BootstrappedPlugin {
+ import net.fabricmc.loom.api.LoomGradleExtensionAPI;
+@@ -83,6 +87,18 @@ public class LoomGradlePlugin implements BootstrappedPlugin {
  		project.apply(ImmutableMap.of("plugin", "java-library"));
  		project.apply(ImmutableMap.of("plugin", "eclipse"));
  
 +		// Tell Gradle that Quilt Loader/Fabric Loader are the same thing
 +		project.getDependencies().getComponents().all(FabricCapabilities.class);
-+
- 		// Setup extensions
- 		project.getExtensions().create(LoomGradleExtensionAPI.class, "loom", LoomGradleExtensionImpl.class, project, LoomFiles.create(project));
- 		project.getExtensions().create("fabricApi", FabricApiExtension.class);
-diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ModConfigurationRemapper.java b/src/main/java/net/fabricmc/loom/configuration/mods/ModConfigurationRemapper.java
-index 009d91ecafc42640135d9ebf5c1eb06be323e115..ac9c0ac012c82a50cb95eb4a9ac8a77613ae255f 100644
---- a/src/main/java/net/fabricmc/loom/configuration/mods/ModConfigurationRemapper.java
-+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ModConfigurationRemapper.java
-@@ -43,6 +43,7 @@ import org.gradle.api.artifacts.Configuration;
- import org.gradle.api.artifacts.FileCollectionDependency;
- import org.gradle.api.artifacts.MutableVersionConstraint;
- import org.gradle.api.artifacts.ResolvedArtifact;
-+import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
- import org.gradle.api.artifacts.dsl.DependencyHandler;
- import org.gradle.api.artifacts.query.ArtifactResolutionQuery;
- import org.gradle.api.artifacts.result.ArtifactResult;
-@@ -131,6 +132,18 @@ public class ModConfigurationRemapper {
- 				configsToRemap.put(entry.getSourceConfiguration().get(), remappedConfig);
- 			}
- 		}
-+		// Round 0: Constraints
-+		// TODO: this is probably the wrong place for this
-+		// Explain to Gradle how to handle conflicts
-+		for (Configuration c : configsToRemap.keySet()) {
++		project.getConfigurations().configureEach(c -> {
 +			// this mirrors how it's done at https://docs.gradle.org/current/userguide/dependency_capability_conflict.html#sub:capabilities for gradle 8.10.2
 +			var rs = c.getResolutionStrategy().getCapabilitiesResolution();
 +			rs.withCapability("net.fabricmc:fabric-loader", a -> {
-+				a.getCandidates().stream().filter(id -> id instanceof ModuleComponentIdentifier && ((ModuleComponentIdentifier) id).getModule().equals("quilt-loader"))
++				a.getCandidates().stream().filter(id -> id.getId() instanceof ModuleComponentIdentifier moduleId && moduleId.getModule().equals("quilt-loader"))
 +						.findFirst().ifPresent(a::select);
 +				a.because("use quilt loader over fabric loader");
```

Comment:
Could calling `because` without any selections cause any issues?

Reply:
The javadoc isn't clear about this, so it's safe to say it's undefined what happens. It's an easy fix, either way, just turn the method reference into a lambda and move the `because` call inside.